From 5dde3af17a784615ab52130702458ca83d5b18bd Mon Sep 17 00:00:00 2001
From: Michael Orlitzky <michael@orlitzky.com>
Date: Mon, 19 Apr 2021 21:11:02 -0400
Subject: [PATCH 1/1] Makefile: support AR and RANLIB variables.

These variables should default to the generic "ar" and "ranlib"
commands, but if the user specifies them we now respect them.

Gentoo-bug: https://bugs.gentoo.org/784230
Gentoo-bug: https://bugs.gentoo.org/784233
---
 Makefile | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 1429643..5742195 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,8 @@
 # Don't edit Makefile! Use conf-* for configuration.
 
 SHELL=/bin/sh
+AR ?= ar
+RANLIB ?= ranlib
 
 default: it
 
@@ -581,7 +583,7 @@ warn-auto.sh systype
 	( cat warn-auto.sh; \
 	echo 'main="$$1"; shift'; \
 	echo 'rm -f "$$main"'; \
-	echo 'ar cr "$$main" $${1+"$$@"}'; \
+	echo '$(AR) cr "$$main" $${1+"$$@"}'; \
 	case "`cat systype`" in \
 	sunos-5.*) ;; \
 	unix_sv*) ;; \
@@ -590,7 +592,7 @@ warn-auto.sh systype
 	dgux-*) ;; \
 	hp-ux-*) ;; \
 	sco*) ;; \
-	*) echo 'ranlib "$$main"' ;; \
+	*) echo '$(RANLIB) "$$main"' ;; \
 	esac \
 	) > makelib
 	chmod 755 makelib
-- 
2.26.3

